README.md

```diff
@@ -139,18 +159,31 @@ $ sw-precache --root=dist --static-file-globs='dist/**/*.html'
 
 The `sw-precache` module exposes two methods: `generate` and `write`.
 
-#### generate(options, callback)
-`generate` takes in [options](#options), generates resulting service worker code as a string, and
-then invokes `callback(error, serviceWorkerString)`.
-In the 1.x releases of `sw-precache`, this was the default and only method exposed by the module.
+#### generate(options, callback) 
+
+`generate` takes in [options](#options), generates a service worker 
+from them and passes the result to a callback fundtion, which must 
+have the following interface:
+
+`callback(error, serviceWorkerString)`
 
-Since 2.2.0, `generate()` also returns a [`Promise`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise).
+In the 1.x releases of `sw-precache`, this was the default and only method
+exposed by the module.
 
-#### write(filePath, options, callback)
-`write` is a helper method that calls `generate` and takes the resulting string content and
-writes it to disk, at `filePath`. It then invokes `callback(error)`.
+Since 2.2.0, `generate()` also returns a
+[`Promise`](https://developer.mozilla.org/en-
+US/docs/Web/JavaScript/Reference/Global_Objects/Promise).
 
-Since 2.2.0, `write()` also returns a [`Promise`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise).
+#### write(filePath, options, callback) 
+`write` takes in [options](#options), generates a service worker from them, 
+and writes the service worker to a specified file. If any errors are 
```

Comment:
`callback(error)` is invoked unconditionally, not just when errors are encountered.
If there are no errors, then the `error` parameter is `null`.
If there is an error, then the `error` parameter contains that error.

The is the standard Node approach to callbacks: http://thenodeway.io/posts/understanding-error-first-callbacks/